### PR TITLE
Only relabelling should apply for scrape_samples_scraped_post_relabel…

### LINKED ISSUE #7298: Samples are dropped due to relabelling, despite the lack of relabelling
Here's query for my custom metrics job...

```
up{endpoint="http", instance="172.17.0.12:9015", job="absolute-epicenter-worker", namespace="default", pod="absolute-epicenter-worker-d4bd57886-hpdmp", service="absolute-epicenter-worker"}
1
scrape_duration_seconds{endpoint="http", instance="172.17.0.12:9015", job="absolute-epicenter-worker", namespace="default", pod="absolute-epicenter-worker-d4bd57886-hpdmp", service="absolute-epicenter-worker"}
0.005390915
scrape_samples_scraped{endpoint="http", instance="172.17.0.12:9015", job="absolute-epicenter-worker", namespace="default", pod="absolute-epicenter-worker-d4bd57886-hpdmp", service="absolute-epicenter-worker"}
16
scrape_samples_post_metric_relabeling{endpoint="http", instance="172.17.0.12:9015", job="absolute-epicenter-worker", namespace="default", pod="absolute-epicenter-worker-d4bd57886-hpdmp", service="absolute-epicenter-worker"}
0
scrape_series_added{endpoint="http", instance="172.17.0.12:9015", job="absolute-epicenter-worker", namespace="default", pod="absolute-epicenter-worker-d4bd57886-hpdmp", service="absolute-epicenter-worker"}
0
count:up1{endpoint="http", job="absolute-epicenter-worker", namespace="default", service="absolute-epicenter-worker"}
1
```

Notice how I have no samples post relabeling? The number scraped is correct. Here's my service monitor. I could fish out the version with the vars filled in, but this is a minikube install with everything in the 'default' namespace, so the values would not be terribly interesting. Both the app label and match label do match the app name, as the app uses exactly same same variables for its name (and release, etc). Other than this I'm relying on the defaults for the prometheus-operator helm chart. Nothing changed. And the monitor works insofar as the number of samples shown for the scrape is correct.

```
---
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: {{ include "epicenter-worker.fullname" . }}
  namespace: {{ .Release.Namespace }}
  labels:
    app: {{ include "epicenter-worker.fullname" . }}
    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
    release: "{{ .Release.Name }}"
    heritage: "{{ .Release.Service }}"
spec:
  selector:
    matchLabels:
      app: {{ include "epicenter-worker.fullname" . }}
      release: "{{ .Release.Name }}"
  namespaceSelector:
    matchNames:
    - {{ .Release.Namespace }}
  endpoints:
  - port: http
    interval: 120s
    path: /epicenter/rest/claxon/prometheus
```

And here's the config from the running Prometheus instance...

```
- job_name: default/absolute-epicenter-worker/0
  honor_timestamps: true
  scrape_interval: 2m
  scrape_timeout: 10s
  metrics_path: /epicenter/rest/claxon/prometheus
  scheme: http
  kubernetes_sd_configs:
  - role: endpoints
    namespaces:
      names:
      - default
  relabel_configs:
  - source_labels: [__meta_kubernetes_service_label_app]
    separator: ;
    regex: absolute-epicenter-worker
    replacement: $1
    action: keep
  - source_labels: [__meta_kubernetes_service_label_release]
    separator: ;
    regex: absolute
    replacement: $1
    action: keep
  - source_labels: [__meta_kubernetes_endpoint_port_name]
    separator: ;
    regex: http
    replacement: $1
    action: keep
  - source_labels: [__meta_kubernetes_endpoint_address_target_kind, __meta_kubernetes_endpoint_address_target_name]
    separator: ;
    regex: Node;(.*)
    target_label: node
    replacement: ${1}
    action: replace
  - source_labels: [__meta_kubernetes_endpoint_address_target_kind, __meta_kubernetes_endpoint_address_target_name]
    separator: ;
    regex: Pod;(.*)
    target_label: pod
    replacement: ${1}
    action: replace
  - source_labels: [__meta_kubernetes_namespace]
    separator: ;
    regex: (.*)
    target_label: namespace
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_service_name]
    separator: ;
    regex: (.*)
    target_label: service
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_name]
    separator: ;
    regex: (.*)
    target_label: pod
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_service_name]
    separator: ;
    regex: (.*)
    target_label: job
    replacement: ${1}
    action: replace
  - separator: ;
    regex: (.*)
    target_label: endpoint
    replacement: http
    action: replace
```

So why am I losing all my samples post relabelling?

### PR DESCRIPTION
**Pull request by brian-brazil**
…ling.

More consistent variable names.

Fixes #7298

### CODE REVIEW

**Comment by roidelapluie** [comment]
Is this a recent regression on master?

**Comment by brian-brazil** [comment]
I didn't dig into when exactly it was introduced, but it is a regression from the original code and intent.

File: scrape/scrape.go [hunk 1]
```diff
@@ -1187,10 +1187,10 @@ loop:
 			}
 		}
 
-		// Increment added even if there's a sampleLimitErr so we correctly report the number of samples scraped.
-		if sampleAdded || sampleLimitErr != nil {
```
Comment (roidelapluie) [inline scrape/scrape.go:1191]: Should we still check if sampleAdded?

**Comment by roidelapluie** [comment]
needs test fix

**Pull request merged**